src/rascal/math/kvec_generator.hh

```diff
@@ -0,0 +1,229 @@
+       * - the Eigen vector of size numvectors containing the norms
+       *   of the found vectors.
+       */
+      size_t numvectors = 0;
+      Matrix_t kvecs{};
+      Vector_t kvecnorms{};
+      // Eigen::Matrix_Xi kvecindices{};
+
+      // Auxiliary variables for developers/testing:
+      size_t numvectors_searchbox{};
+      size_t n1max{};
+      size_t n2max{};
+      size_t n3max{};
+      bool include_origin{};
+      Eigen::Matrix3d innerprodmatrixM{};
+      double detM{};
+
+     public:
+      /** Constructor:
+       * As part of the constructor, the precompute() function is
+       * automatically called which computes the k-vectors for the
+       * provided parameters.
+       * @param cutoff Cutoff radius of reciprocal space ball.
+       *     As a guideline, if the smearing parameter of the density
+       *     is given by \f$sigma\f$, use \f$pi/sigma\f$ as cutoff.
+       * @param basisvectors 3x3 matrix containing the three basis
+       *     vectors a1, a2, a3 of the real space unit cell, stored as:
+       *     a1 = basisvectors[0], a2=basisvectors[1], etc.
+       *     These are then automatically converted to the corresponding
+       *     reciprocal cell vectors b1, b2, b3 by the formulae:
+       *     \f{equation}{b_1 = 2\pi \frac{a_1 \times a_2}
+       *                 {a_1 \cdot (a_2 \times a_3)} \f}
+       *     and similarly for the periodic permutations.
+       *     It is also possible to provide the basis vectors of the
+       *     reciprocal lattice directly by using the option below.
+       * @param is_reciprocal_cell false by default.
+       *     If set to true, it indicates that the cell provided
+       *     in the variable basisvectors is already the reciprocal cell.
+       *     Thus, the formula above for converting the a_i's to the b_i's
+       *     is not used, and the provided cell is used directly.
+       * @param need_origin false by default.
+       *     If set to true, the list of found k-vectors will also
+       *     contain the origin k=(0,0,0), which will then be stored
+       *     as the very first element in the matrix of found vectors.
+       */
+      explicit Kvectors(double cutoff, Eigen::Matrix3d basisvectors,
+                        bool is_reciprocal_cell = false,
+                        bool need_origin = false) {
+        // Set internal variables apart from cell
+        this->kcutoff = cutoff;
+        this->include_origin = need_origin;
+
+        /* Store provided basis vectors into the internal variable basisvecs.
+         * We need to distinguish whether the provided vectors are already
+         * reciprocal cell vectors or need to be converted first
+         */
+        if (not is_reciprocal_cell) {  // real space cell is given
+          // Create reciprocal cell from real space cell
+          Eigen::Matrix3d tcell = basisvectors.transpose();
+          this->basisvecs = 2.0 * M_PI * tcell.inverse();
```

Comment:
I don't know why, but we have defined our own pi constant in math/utils.hh I would just use it here for consistency within the project